File: frontend/src/schemas/appointment.ts
import { z } from "zod";

export const createAppointmentSchema = z.object({
  dateTime: z
    .date({ error: "Date and time are required" })
    .refine((val) => val > new Date(), "Appointment should be in the future"),
  dentistId: z.string().min(1, "Please select a dentist"),
  treatmentId: z.string().min(1, "Please select a treatment"),
});

export type CreateAppointmentFormData = z.infer<typeof createAppointmentSchema>;

File: backend/Entities/Appointment.cs
namespace backend.Entities;

public class Appointment
{
    public Guid Id { get; set; }
    public DateTime ScheduledAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Guid PatientId { get; set; }
    public Patient Patient { get; set; } = null!;

    public Guid DentistId { get; set; }
    public Dentist Dentist { get; set; } = null!;

    public Guid TreatmentId { get; set; }
    public Treatment Treatment { get; set; } = null!;
}